Src/FluentAssertions/Formatting/FormattedObjectGraph.cs

```diff
@@ -45,84 +63,72 @@ public FormattedObjectGraph(int maxLines)
     public void AddFragmentOnNewLine(string fragment)
     {
         FlushCurrentLine();
-
-        AddFragment(fragment);
+        GetCurrentLine().Append(fragment);
     }
 
     /// <summary>
-    /// Starts a new line with the provided line of text that does not allow
-    /// adding more fragments of text.
+    /// If there's only one line, adds a fragment to that line. If there are more lines, adds the fragment as
+    /// a new line that does not allow any further fragments.
     /// </summary>
-    public void AddLine(string line)
+    public void AddLineOrFragment(string fragment)
     {
-        FlushCurrentLine();
-
-        AppendWithoutExceedingMaximumLines(Whitespace + line);
-    }
-
-    /// <summary>
-    /// Adds a new fragment of text to the current line.
-    /// </summary>
-    public void AddFragment(string fragment)
-    {
-        if (lineBuilderWhitespace.Length > 0)
+        if (lines.Count == 1)
         {
-            lineBuilder.Append(lineBuilderWhitespace);
-            lineBuilderWhitespace = string.Empty;
+            AddFragment(fragment);
+        }
+        else
+        {
+            AddLine(fragment);
         }
-
-        lineBuilder.Append(fragment);
     }
 
     /// <summary>
-    /// Adds a new line if there are no lines and no fragment that would cause a new line.
+    /// Starts a new line with the provided text that does not allow adding more
+    /// fragments of text.
     /// </summary>
-    internal void EnsureInitialNewLine()
+    public void AddLine(string content)
     {
-        if (LineCount == 0)
-        {
-            InsertInitialNewLine();
-        }
+        FlushCurrentLine();
+
+        GetCurrentLine().Append(content);
+        FlushCurrentLine();
     }
 
     /// <summary>
-    /// Inserts an empty line as the first line unless it is already.
+    /// Adds a new fragment of text to the current line.
     /// </summary>
-    private void InsertInitialNewLine()
+    public void AddFragment(string fragment)
     {
-        if (lines.Count == 0 || !string.IsNullOrEmpty(lines[0]))
-        {
-            lines.Insert(0, string.Empty);
-            lineBuilderWhitespace = Whitespace;
-        }
+        GetCurrentLine().Append(fragment);
     }
 
     private void FlushCurrentLine()
     {
-        string line = lineBuilder.ToString().TrimEnd();
-        if (line.Length > 0)
+        // We only need to flush the line if there's something to flush.
+        if (currentLine is not null)
         {
-            AppendWithoutExceedingMaximumLines(lineBuilderWhitespace + line);
+            currentLine.Flush();
```

Comment:
🙃 Seems we don't need to call `Flush()` explicitly, only starting a new line by setting `currentLine = null`.

Reply:
You don't _have_ to, but it won't close off the builder for that line.